src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -91,20 +92,46 @@ public void setupSchema(List<ColumnMetadata> columns) {
       if (!column.getNullable()) {
         addNonNullableFieldName(column.getInternalName());
       }
-      this.statsMap.put(
-          column.getInternalName(),
-          new RowBufferStats(column.getName(), column.getCollation(), column.getOrdinal()));
-
-      if (onErrorOption == OpenChannelRequest.OnErrorOption.ABORT
-          || onErrorOption == OpenChannelRequest.OnErrorOption.SKIP_BATCH) {
-        this.tempStatsMap.put(
+      if (!clientBufferParameters.getIsIcebergMode()) {
```

Comment:
add comment on why the stats map tracking is done differently, and whether there's any qualitative differences. For example its column.getInternalName here and columnDescriptor.getPath() in iceberg mode.

Reply:
Added comments. Actually streaming to FDN can also use the below logic of setting up Iceberg stats. The `column.getInternalName` is the same as `columnDescriptor.getPath()` as the parquet column was set up using internal name.